### PR TITLE
chore: update integration test commands to allow selection

### PR DESCRIPTION
**Pull request by Jordan-Nelson**
*Issue #, if available:*

*Description of changes:*
- update `test:integration:ios/android` to run for a single package (with a prompt) by default, with the ability to override with `--no-select`
- update `test:integration` to use the `--no-select` package so that `test:integration` still runs tests all packages
- update the contributing docs

The melos commands for integ tests run tests for all packages by default. There is no easy way to choose a single package. This update makes running the tests for a single package & platform easier, while still allowing them to be run on for all packages with the option `--no-select`.

Note: Previously the melos commands were updated to *not* use prompts (https://github.com/aws-amplify/amplify-flutter/pull/711) in order to make running them all at once easier. However `--no-select` makes this pretty easy as well.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -154,45 +154,57 @@ toolkit for interacting with AWS backend resources.
 
 ## Integration Tests
 
-In addition to unit tests which mock Amplify API interaction, this repository has integration tests which 
-test functionality with real Amplify backends. The integration test script will execute tests in example 
+In addition to unit tests which mock Amplify API interaction, this repository has integration tests which
```
Comment (Jordan-Nelson) [inline CONTRIBUTING.md:157]: The formatting changes in this file are auto formatting from prettier. The changes generally seem to be an improvement, and the majority of the file was already following that formatting, so I decided to leave them.

Pro tip: use https://github.com/aws-amplify/amplify-flutter/pull/910/files?diff=split&w=1 to ignore the whitespace changes

I also updated a few typos in the file.

**Review by dnys1 — APPROVED** [review]

**Review by HuiSF — APPROVED** [review]
Thanks for the updates!

File: melos.yaml [hunk 2]
```diff
@@ -39,16 +39,31 @@ scripts:
     ./.circleci/test_all_plugins.sh ios-test && exit 0
 
   test:integration:
-    run: melos run test:integration:android && melos run test:integration:ios
+    run: melos run test:integration:android --no-select && melos run test:integration:ios --no-select
```
Comment (ragingsquirrel3) [inline melos.yaml:42]: Cool! Did not know about `no-select` option.

File: melos.yaml [hunk 3]
```diff
@@ -39,16 +39,31 @@ scripts:
...
     description:
       Run all integration tests for all package example apps on Android and iOS simulators. Skips if no tests available.
       - Requires running Android and iOS simulators.
 
   test:integration:android:
-    run: melos exec -c 1 --scope="*example*" --file-exists="integration_test/main_test.dart" "flutter drive --no-pub --driver=test_driver/integration_test.dart --target=integration_test/main_test.dart -d sdk"
+    run: melos exec -c 1 "flutter drive --no-pub --driver=test_driver/integration_test.dart --target=integration_test/main_test.dart -d sdk"
```
Comment (ragingsquirrel3) [inline melos.yaml:48]: Thanks for restoring this syntax!

**Review by ragingsquirrel3 — APPROVED** [review]
Thanks! Makes sense.

**Pull request merged**